File: recover.js
const fs = require('fs');

const logPath = 'C:/Users/naldo/.gemini/antigravity/brain/addb1d88-f35b-4fc1-a009-23b6dddaec2c/.system_generated/logs/overview.txt';

if (fs.existsSync(logPath)) {
    console.log('Log exists, reading...');
    const txt = fs.readFileSync(logPath, 'utf8');
    
    // Procura o último bloco do multi_replace_file_content e o respectivo TargetContent
    const identifier = '"TargetContent":"            // 1. Traz Ordem';
    const lastIdx = txt.lastIndexOf(identifier);
    
    if (lastIdx !== -1) {
        console.log('Found block!');
        const start = lastIdx + 17; // Pula "TargetContent":"
        const endIndicator = '","StartLine"';
        let end = txt.indexOf(endIndicator, start);
        if(end === -1) {
             end = txt.indexOf('"}', start);
        }
        
        let content = txt.substring(start, end);
        console.log('Raw content size:', content.length);
        
        // Fix string escape characters JSON encoded inside log
        content = content.replace(/\\n/g, '\n').replace(/\\"/g, '"');
        
        fs.writeFileSync('C:/ecossistema arnaldo trentin/frontend/js/app.js.recovered', content);
        console.log('RECOVERED OK! Length:', content.length);
    } else {
        console.log('Target block not found!');
        
        // Fallback: search just for "TargetContent" and log surrounding text
        const tIdx = txt.lastIndexOf('TargetContent');
        if (tIdx > -1) console.log(txt.substring(tIdx - 50, tIdx + 100));
    }
} else {
    console.error('File not found:', logPath);
}

File: recover.py
import json, sys

log_path = r'C:\Users\naldo\.gemini\antigravity\brain\addb1d88-f35b-4fc1-a009-23b6dddaec2c\.system_generated\logs\overview.txt'

with open(log_path, 'r', encoding='utf-8') as f:
    text = f.read()

# Ache todas as chamadas "multi_replace_file_content" na conversa
idx = text.rfind('multi_replace_file_content')
if idx == -1:
    print('Block not found!')
    sys.exit(1)

# Encontre a chamada contendo "1725" (linhas deletadas na interface) ou TargetContent no formSuperOS
try:
    # A string do TargetContent está entre as chamadas tool... vamos varrer as chaves json "TargetContent"
    # Mas overview.txt não contém JSON perfeitamente formatado para json.loads, é dump da conversa.
    # Vamos buscar por substring exata: '"TargetContent":"            // 1. Traz Ordem de Serviço'
    
    start_str = '"TargetContent":"            // 1. Traz Ordem'
    start = text.rfind(start_str)
    if start == -1:
        start_str = r'"TargetContent":"            // 1. Traz Ordem'
        start = text.rfind(start_str)
        if start == -1:
            print("Target content string not found")
            sys.exit(1)

    end_str = '"}'
    end = text.find(end_str, start)
    
    val_str = text[start+17:end]
    
    # Eval para traduzir os escapes \n
    import ast
    content = ast.literal_eval('"' + val_str.replace('"', '\\"') + '"')
    
    with open('RECOVERED.js', 'w', encoding='utf-8') as out:
        out.write(content)
        
    print(f'Recovered {len(content)} characters!')
except Exception as e:
    print(e)
